### ISSUE TITLE
Some emoji are not supported

### ISSUE THREAD

**Issue by Gaboris** [getferdi/ferdi#96]
**Describe the bug**
Within services like Telegram some emoji are not supported and just display the default "missing" rectangle, but even in an old browser some more get supported and up to date one supports almost all of them so it doesn't seem like a system wide bug.

I was hoping that this is just an outdated Electron v4 problem and was hoping updating that would fix it, but since we are already on v6 and it's still a thing I guess it's a separate module.

**To Reproduce**
Just open a service that doesn't seem to supply its own font/character sets like Telegram and look for specialized, later implemented or rare emoji like Kiwi or Black heart that doesn't get displayed.

**Desktop (please complete the following information):**

- OS: Windowns 7 Ultimate 64bit Hungarian
- Browser: Firefox, Chrome, Opera, Vivaldi, Internet Explorer
- Version: Since launch of Franz

**Comment by Gaboris** [comment]
Sorry, there were supposed to be images as well.

![ferdi](https://user-images.githubusercontent.com/9462372/66035949-6ba73800-e50c-11e9-9209-33bd04800b79.jpg)
![firefox56](https://user-images.githubusercontent.com/9462372/66035950-6ba73800-e50c-11e9-9cb1-71c0e95f6ff8.jpg)
![firefox69](https://user-images.githubusercontent.com/9462372/66035952-6ba73800-e50c-11e9-85a2-c0fad80ac635.jpg)

**Comment by AGCaesar** [comment]
I never use emoji so maybe I am not the best person to test...
On my phone I can search for emojis and send them, on the web version this does not seem possible... but they all get displayed (well the black heart and the kiwi I tied :D)

But I am on linux, maybe that makes a difference..

**Comment by Gaboris** [comment]
Yeah neither do I, but people around me do and it bugs me. :D

That's the other thing.
Web based Telegram uses the "system" font that's within the browser running it. Maybe Linux just uses external repositories(as far as I can remember Linux apps get kinda compiled on install) that get updated fonts while on Windows it might have something built in that has to be updated within the code or just grabs what the OS has installed in which case I need to figure out how to update that.

That's the actual question I'm looking for I guess.

**Comment by AGCaesar** [comment]
Ah... well on Linux yyou need to have a font supporting emojis to display them... Sorry I cant help with windows, not even sure which font you would install...
Maybe give this one a try?
https://www.google.com/get/noto/help/install/

**Comment by Gaboris** [comment]
Thanks for the heads up. Gonna try it if I get some feedback on this in case it's not all based within Ferdi. :)

**Comment by vantezzen** [comment]
Alternatively, we could think about adding a library like [twemoji](https://github.com/twitter/twemoji) to add better emoji support for some platforms.

**Comment by ruru4143** [comment]
Had the same issue and solved it by installing a emoji font (on linux)

**Comment by mhatvan** [comment]
Cannot reproduce this with `telegram-react` recipe on Ferdi 5.6.1-nightly.33

I do remember that there was an issue with emoji fonts in Chromium a while ago on Ubuntu, but unrelated to Ferdi.

Please comment in case this issue persists, then I will reopen.

**Comment by RJVB** [comment]
There's something of an issue with the kind of emojis that get used in different services. For me the ones I see in FB Messenger  and Whatsapp are the same colour emoji as I get in a browser and on my phone.
With Instagram, I get monochrome emoji, despite having the `Noto Colour Emoji` font installed; in (Firefox-based) browsers and on my phone I get the same kind of coloured glyphs as in Messenger and Whatsapp.

Some of my correspondents make it an art to condense (maybe not) a thousand words in deliberately chosen images and -I feel a bit silly admitting this- I sometimes find it difficult to tell certain monochrome emoji apart.

But maybe this can be handled via CSS?

**Comment by vraravam** [comment]
Ferdi is only rendering the webpages of each service in a browser as a tab. And Ferdi does not have the ability for the user to specify their own fonts. So, I'm not sure whether your installed fonts can affect what Ferdi shows.

**Comment by RJVB** [comment]
Well, web pages that don't load their fonts from somewhere off the interweb are inevitably going to be affected by what fonts we have installed, and that's undoubtedly the reason why fonts specifications tend to be lists rather than single typeface names.

The question remains whether this is something that can be overridden through CSS (or the user.js) For regular fonts it's almost certainly possible (FBPurity can change the fonts used by Facebook that way). 
The YakYak hangouts client has this in its `app.css` file:

```
...
@font-face {
  font-family: 'NotoColorEmoji';
  src: url('./css/noto-color-emoji/NotoColorEmoji.ttf') format('truetype');
  font-weight: normal;
  font-style: normal;
}
...
html {
  --emojifont: Roboto, "Helvetica Neue", NotoColorEmoji, 'Segoe UI Emoji', Helvetica, Arial, "Lucida Grande", sans-serif;
  --globalfont: Roboto, "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;
}
...
img.emoji {
  height: 1em;
  width: 1em;
  margin: 0 0.05em 0 0.1em;
  vertical-align: -0.1em;
}
....
.convlist .conv .convinfos .lastmessage .message {
  font-family: var(--emojifont);
}
...
.input {
  font-family: var(--emojifont);
  position: relative;
}
...
.messages .ugroup .umessages .message {
  font-family: var(--emojifont);
  padding: 5px;
  white-space: pre-wrap;
  word-break: break-word;
}
```

I'd experiment (also with the user agent), but can't right now ... It's as if the entire Facebook universe has disappeared (even the WHOIS info) :-/

**Comment by RJVB** [comment]
Here's an interesting observation: neither Google Chrome nor Electron-based applications are affected by the "monochrome emoji" issue on ChromeOS. It's one thing that the official platform browser could have tweaks in this aspect, but the nodejs-based software and Electron running on my chromebook are basically the Linux binaries, and not even the latest versions (Electron and nodejs 12.x installed through *crew* and *nodebrew*).

**Comment by vraravam** [comment]
Electron packages *Chromium* (not Chrome).

Also, just this morning, I had committed and merged the latest version of electron `15.1.1`. Since the signing process is currently broken, this has not yet been released as a nightly. Waiting for @kytwb to unblock us.

**Comment by RJVB** [comment]
>Electron packages *Chromium* (not Chrome).

So does (Google) Chrome (I'm confident that Electron also adds its own tweaks to pure Chromium). Either way, all Chromium-based browsers display monochrome emoji in Instagram on Linux (and Mac). The fact that this is not the case on ChromeOS suggests a difference at the OS level, FOR INSTANCE in terms of what fonts are installed. Annoyingly I have no idea how to check what font the browser uses for (colour) emoji.
